### PR TITLE
Eval logprob benchmarks support for HF in-fw with Ray

### PR DESCRIPTION
**Pull request by athitten**
This PR adds changes required for OAI API compatibility for HF in-fw deployment with Ray to be able to run evaluation logprob benchmarks. These are the major changes:

- `generate` method in `nemo_deploy/llm/hf_deployable.py`: Adds `echo` arg to the generate method and adds logic to retrieve prompt + generated tokens or just  generated tokens accordingly.
- `ray_infer_fn` method in `nemo_deploy/llm/hf_deployable.py`: Adds args `compute_logprob`, `n_top_logprobs`, `echo` to `ray_infer_fn` for OAI API compatibility and adds code to get `logprobs` and `top_logprobs` for generated tokens (from the output of generate method) and prompt tokens (by running fwd pass on the model with prompt tokens to get their logits).
- `completions` endpoint in `nemo_deploy/llm/hf_deployable_ray.py`: Exposes logprobs, echo args as incoming eval requests for logprob benchmarks contain them and adds some post processing to the final output for OAI API compatibility.
- Adds number of unit tests for these changes and also a new test file for explicitly testing the OAI API compatibility: `tests/unit_tests/deploy/test_hf_ray_oai_format.py`

 Accuracy on Llama 3.2 1B:
 
 | LogProb Benchmarks | HF in-fw with Ray | Baseline |
| -------- | -------- | -------- |
| arc_challenge   |   36.69%   |   36.60%  |
| openbookqa  |    37.4%  | 38.8%  |
| truthfulqa_mc2 |    38.47%  | 38.4%  |
| piqa  |    74.8%  | 74.92%  |

### CODE REVIEW

File: nemo_deploy/llm/hf_deployable.py [hunk 1]
```diff
@@ -370,38 +389,154 @@ def ray_infer_fn(self, inputs: Dict[Any, Any]):
...
+                        truncation=self.tokenizer_truncation,
+                    )
+                    prompt_token_ids = prompt_inputs["input_ids"]
+                    # Move to same device as model
+                    for key, val in prompt_inputs.items():
+                        if torch.is_tensor(val):
+                            prompt_inputs[key] = val.cuda()
+
+                # Process each sample
+                log_probs_list = []
```
Comment (oyilmaz-nvidia) [inline nemo_deploy/llm/hf_deployable.py]: Do you think it makes sense to move this part in a separate function and call it here? Do we also do this for PyTriton?

Comment (athitten) [reply nemo_deploy/llm/hf_deployable.py]: Actually its good to move it to a separate function for better readability and clean code, thanks @oyilmaz-nvidia. We don't need this for PyTriton as we are supporting evaluation of HF models with just Ray.

Let me move this.
